manifestival.go

```diff
@@ -131,9 +131,9 @@ func (m Manifest) apply(spec *unstructured.Unstructured, opts ...ApplyOption) er
 	}
 	if current == nil {
 		m.logResource("Creating", spec)
+		annotate(spec, v1.LastAppliedConfigAnnotation, lastApplied(spec))
+		annotate(spec, "manifestival", resourceCreated)
 		current = spec.DeepCopy()
-		annotate(current, v1.LastAppliedConfigAnnotation, lastApplied(current))
-		annotate(current, "manifestival", resourceCreated)
```

Comment:
We don't want to do this. By mutating the spec instead of its deep copy, you're actually changing the manifest as it was loaded. Effectively what you're doing to solve your problem is tricking manifestival into thinking it created the namespace when in fact it may not have.

Is it possible you applied resources to an existing namespace? We don't want manifestival deleting any namespace it didn't actually create.

Reply:
The `namespace` is created by manifestival, I'm sure : ), 
I agree it actually changing the manifest as it was loaded, any good suggestion?

Reply:
Ok, now I see what you're saying! I think it's fine to update the spec, but only with the resourceCreated annotation, not the last-applied. We used to do update every field in the spec, which is why I was confused it wasn't working, but other problems result from doing that. I don't think just that one annotation will be a problem, though.

Thanks!

Reply:
Thanks a lot, happy to work with you.